scripts/release-bit-verification.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/sh
+
+############################################################################
+#  PREREQUISITES FOR THIS SCRIPT
+# 1. Redistributable-binary(.rpm) should be passed as the first argument
+# 2. Login to the cluster should be done prior to running this script
+# 3. The cluster should be in a state where it can be used for testing
+#
+# USAGE:
+# ./release-bit-verification.sh redistributable-binary
+#
+# Example: ./release-bit-verification.sh ~/Downloads/odo-redistributable-2.4.3-1.el8.x86_64.rpm
+#
+
+shout() {
+    echo "--------------------------------$1------------------------------------------"
+}
+# Check SHASUM for all the binary files and there should be no difference
+
+# Create a Temp directory
+WORKING_DIR=$(mktemp -d)
+shout "WORKING_DIR=$WORKING_DIR"
+export REPO_URL=${REPO_URL:-"https://github.com/redhat-developer/odo.git"}
+
+# Extract from rpm file
+rpm2cpio ${1} | cpio -idmvD $WORKING_DIR
+pushd $WORKING_DIR/usr/share/odo-redistributable/
+
+# Check sha256sum for all the files
+while IFS= read -r line; do
+    read -r SHA FILE <<<"$line"
+    read -r SHATOCHECK FILE <<<$(sha256sum $FILE)
+    if [[ $SHA == $SHATOCHECK ]]; then
+        # Print if the file is correct
+        printf '%-50s\U0002705\n' $FILE
+    fi
+done <SHA256_SUM
+
+shout
+
+# Copy binary for testing purpose
+OS=$(uname -s)
+ARCH=$(uname -m)
+
+if [[ $OS == "Linux" ]]; then
+    if [[ $ARCH == "x86_64" ]]; then
+        cp ./odo-linux-amd64 odo
+        PATH=$(pwd):$PATH
+    fi
+fi
+
+# Check odo verion and if it is correct
+VERSION=$(cat VERSION)
+ODOVERSIONCHECK=$(odo version)
+if [[ "$ODOVERSIONCHECK" == *"$VERSION"* ]]; then
+    echo "odo binary is installed correctly"
+fi
```

Comment:
What if there's version mismatch?

Reply:
Well, In that case we should stop the test.
I will add a failure condition to stop the test with failure.
Thanks Dharmit, your question was helpful.